### PR TITLE
Navigation ryan us4 5 8 9

### PR DESCRIPTION
**Pull request by cmpprg**
- Made tests for everything below.
- Added specific navigation for specific users.
-created 404 restrictions for specific pages

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,12 +1,10 @@
 class User < ApplicationRecord
-  validates :name, presence:true
-  validates :address, presence:true
-  validates :city, presence:true
-  validates :state, presence:true
-  validates :zip, presence:true
+  validates :name, :address, :city, :state, :zip, presence:true
   validates :email, presence:true, uniqueness:true
-  validates :password, presence:true
+  validates_presence_of :password, required:true
```
Comment (tony-gomes) [inline app/models/user.rb:4]: this is so tricky that it needs the whole call...

File: app/models/user.rb [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
   validates :password_confirmation, presence:true
 
+  enum role: { regular: 0, merchant: 1, admin: 2 }
```
Comment (tony-gomes) [inline app/models/user.rb:7]: great find & super clean...

File: app/controllers/admin/admin_controller.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Admin::AdminController < ApplicationController
+  def show
+    render file: "/public/404" unless current_admin?
+  end
+end
```
Comment (tylerpporter) [inline app/controllers/admin/admin_controller.rb]: We will end up adopting the BaseController and this logic will live there.

**Review by tony-gomes — APPROVED** [review]
Great work!

File: spec/features/admin_user/nav_access_spec.rb [hunk 4]
```diff
@@ -0,0 +1,110 @@
+require "rails_helper"
+
+RSpec.describe "As a admin level user" do
```
Comment (Shock253) [inline spec/features/admin_user/nav_access_spec.rb:3]: It looks like these test files are organized differently than our tests, but there shouldn't be any problems

**Review by tylerpporter — APPROVED** [review]

**Review by Shock253 — APPROVED** [review]

**Pull request merged**